src/Core/DataBlock/DataBlock.cc

```diff
@@ -608,10 +639,54 @@ static bool ConvertDataTypeInternal( DATA* src, DataBlockHandle& dst_data_block
       for ( ; j < size; j++ )
       {
         dst[ j ] = static_cast<unsigned int>( src[ j ] );
-      } 
+      }
+      return true;
+    }
+    case DataType::LONGLONG_E:
+    {
+      long long* dst = reinterpret_cast<long long*>( dst_data_block->get_data() );
+      size_t size8 = size & ~(0x7);
+      size_t j = 0;
+      for ( ; j < size8; j+=8 )
+      {
+        dst[ j ] = static_cast<long long>( src[ j ] );
+        dst[ j + 1 ] = static_cast<long long>( src[ j + 1 ] );
+        dst[ j + 2 ] = static_cast<long long>( src[ j + 2 ] );
+        dst[ j + 3 ] = static_cast<long long>( src[ j + 3 ] );
+        dst[ j + 4 ] = static_cast<long long>( src[ j + 4 ] );
+        dst[ j + 5 ] = static_cast<long long>( src[ j + 5 ] );
+        dst[ j + 6 ] = static_cast<long long>( src[ j + 6 ] );
+        dst[ j + 7 ] = static_cast<long long>( src[ j + 7 ] );
+      }
```

Comment:
This loop seems unnecessary. I don't think unrolling nets any performance here because a loop iteration does not need to wait on the result of the previous iteration.

Reply:
This sort of thing should be profiled. I agree it seems like an antiquated form of optimization, but without testing, how will we know the impact? I think since it's Datablock copy code, it should be easy to test--just load up a few big layers and copy them around.

Reply:
This refactoring is not quite related to migration to ITKv5. Can we move it to a separate discussion? Or if someone confirms there is no difference or it is faster with this commit, merge it as part of this PR?

Reply:
Yeah I shouldve jumped in and suggested that after Tark put in his unrelated suggestions and before you added the reply. Let's have you back out that commit and stash it on a separate branch.

Reply:
Sorry, I should have made this comment elsewhere.